docs/development/authentication.md

```diff
@@ -0,0 +1,137 @@
+
+## Authentication for Git
+
+There are two ways for authenticating into Git(_applies to both Gitlab or Github_). Also, see more information in the official Tekton [docs](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#configuring-ssh-auth-authentication-for-git).
+
+### SSH authentication
+
+For the ssh authentication you can use the tekton annotations to specify if you are cloning from github or gitlab. Is recommended to always use both when defining your secret.
+
+As seen in the following example, there are three things to notice:
+
+- The Kubernetes secret should be of the type `kubernetes.io/ssh-auth`
+- The annotations include a value to support both github and gitlab
+- The `data.ssh-privatekey` can be generated by following the command example `cat ~/.ssh/id_rsa | base64`, where `~/.ssh/id_rsa` is the key used to authenticate into Git.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-git-ssh-auth
+  annotations:
+    tekton.dev/git-0: github.com
+    tekton.dev/git-1: gitlab.com
+type: kubernetes.io/ssh-auth
+data:
+  ssh-privatekey: <cat ~/.ssh/id_rsa | base64>
+```
+
+### Basic authentication
+
+The Basic authentication is very similar to the ssh one, but with the following differences:
+
+- The Kubernetes secret should be of the type `kubernetes.io/basic-auth`
+- The `stringData` should host your user and password in clear text.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-git-basic-auth
+  annotations:
+    tekton.dev/git-0: https://github.com
+    tekton.dev/git-1: https://gitlab.com
+type: kubernetes.io/basic-auth
+stringData:
+  username: <cleartext username>
+  password: <cleartext password>
+```
+
+### Usage of git secret
+
+With the right secret in place(_Note: Ensure creation of secret in the proper Kubernetes namespace_), users should reference it on their Build YAML definitions.
+
+Depending on the secret type, there are two ways of doing this:
+
+When using ssh auth, users should follow:
+
+```yaml
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: buildah-golang-build
+spec:
+  source:
+    url: git@gitlab.com:eduardooli/newtaxi.git
+    credentials:
+      name: secret-git-ssh-auth
+```
+
+When using basic auth, users should follow:
+
+```yaml
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: buildah-golang-build
+spec:
+  source:
+    url: https://gitlab.com/eduardooli/newtaxi.git
+    credentials:
+      name: secret-git-basic-auth
+
+```
+
+## Authentication to container registries
+
+For pushing images to private registries, users require to define a secret in their respective namespace.
+
+### Docker Hub
+
+Follow the following command to generate your secret:
+
+```sh
+kubectl -n <YOUR_NAMESPACE> create secret docker-registry <CONTAINER_REGISTRY_SECRET_NAME> --docker-server=<REGISTRY_HOST> --docker-username=<USERNAME> --docker-password=<PASSWORD> --docker-email=me@here.com
```

Comment:
Just as a multi-line command for readability.
```suggestion
kubectl --namespace <YOUR_NAMESPACE> create secret docker-registry <CONTAINER_REGISTRY_SECRET_NAME> \
  --docker-server=<REGISTRY_HOST> \
  --docker-username=<USERNAME> \
  --docker-password=<PASSWORD> \
  --docker-email=me@here.com
```

Reply:
make sense, done